test/integration/targets/ansible-doc/test.yml

```
@@ -10,6 +10,37 @@
     actual_output_clean: '{{ actual_output.splitlines()[1:] }}'
     expected_output_clean: '{{ expected_output.splitlines()[1:] }}'
   tasks:
+    - name: create temporary file
+      ansible.builtin.tempfile:
+        state: file
+      register: temp_file
+
+    - name: redirect stdout and stderr to file
+      ansible.builtin.shell: |
+        ansible-doc test_docs &> {{ temp_file.path }}
+      args:
+        executable: "{{ ansible_facts.user_shell }}"
```

If the issue is the `bash`-style redirection you're using, you should be able to just use the standard POSIX equivalent.
```suggestion
      ansible.builtin.shell:
        cmd: ansible-doc test_docs > {{ temp_file.path }} 2>&1
```

---

Thanks